diff --git a/drivers/net/ethernet/mellanox/mlx5/core/en_main.c b/drivers/net/ethernet/mellanox/mlx5/core/en_main.c
--- a/drivers/net/ethernet/mellanox/mlx5/core/en_main.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_main.c
@@ -7070,2 +7070,0 @@ static const struct netdev_queue_mgmt_ops mlx5e_queue_mgmt_ops = {
-	.ndo_validate_qcfg	=	mlx5e_queue_validate_qcfg,
-	.supported_params       =	QCFG_RX_PAGE_SIZE,
